Process.c

```diff
@@ -121,35 +117,42 @@ static int matchCmdlinePrefixWithExeSuffix(const char* cmdline, int cmdlineBaseO
     * that make htop's identification of the basename in cmdline unreliable.
     * For e.g. /usr/libexec/gdm-session-worker modifies its cmdline to
     * "gdm-session-worker [pam/gdm-autologin]" and htop ends up with
-    * proccmdlineBasenameEnd at "gdm-autologin]". This issue could arise with
+    * cmdlineBasenameStart at "gdm-autologin]". This issue could arise with
     * chrome as well as it stores in cmdline its concatenated argument vector,
     * without NUL delimiter between the arguments (which may contain a '/')
     *
     * So if needed, we adjust cmdlineBaseOffset to the previous (if any)
     * component of the cmdline relative path, and retry the procedure. */
-   bool delimFound; /* if valid basename delimiter found */
+   size_t cmdlineBaseOffset = *cmdlineBasenameStart;
+   bool delimFound = true; /* if valid basename delimiter found */
    do {
       /* match basename */
       matchLen = exeBaseLen + cmdlineBaseOffset;
       if (cmdlineBaseOffset < exeBaseOffset &&
           strncmp(cmdline + cmdlineBaseOffset, exe + exeBaseOffset, exeBaseLen) == 0) {
-         delim = cmdline[matchLen];
+         char delim = cmdline[matchLen];
          if (delim == 0 || delim == '\n' || delim == ' ') {
-            int i, j;
+            size_t i, j;
             /* reverse match the cmdline prefix and exe suffix */
             for (i = cmdlineBaseOffset - 1, j = exeBaseOffset - 1;
```

Comment:
This could underflow for both `i` and `j` …

Reply:
Intentional. `i` and `j` may wrap over to `(size_t)-1` and that's what the conditionals below are for.

Reply:
Don't quite like doing it that way …

Especially as `x >= 0` is a no-op (always `true`) for any unsigned `x` …

Reply:
Do you have any better way here? Note that the `i >= 0` had been rewritten to `i != (size_t)-1` by me when I changed the types to unsigned.

Reply:
Update: I think I can change this one a little bit. Just realized the checks against `(size_t)-1` here are too ugly, and can be avoided if we have the `i` and `j` counters incremented by 1. A decent compiler should optimise it to the same code (and if it doesn't it's a bug).

```c
            for (size_t i = cmdlineBaseOffset, j = exeBaseOffset;
               i != 0 && j != 0 && cmdline[i - 1] == exe[j - 1];
               --i, --j)
               /* empty loop body */;
            if (i == 0 && j != 0 && exe[j - 1] == '/') {
               *cmdlineBasenameStart = cmdlineBaseOffset;
               return matchLen;
            }